t/10-jobs.t

```diff
@@ -689,13 +690,45 @@ subtest 'carry over, including soft-fails' => sub {
             $job_info = $t->app->minion->jobs({tasks => ['hook_script']})->next;
             is $job_info->{state}, 'inactive', 'hook script has been retried with long delay';
             is_deeply($job_info->{args}[2],
-                {delay => 60, retries => 1440, skip_rc => 142, kill_timeout => '30s', timeout => '5m'});
+                {delay => 60, retries => $default_retries, skip_rc => 142, kill_timeout => '30s', timeout => '5m'});
             $notes = $job_info->{notes};
             is($notes->{hook_cmd}, 'echo delayed && exit 142;', 'real hook cmd in notes if result matches (5)');
             like($notes->{hook_result}, qr/delayed/, 'real hook cmd from config called if result matches (5)');
             is $notes->{hook_rc}, 142, 'exit code of the hook cmd is as expected';
             is $job_info->{retries}, 1, 'hook script has been retried once because of delay';
         };
+
+        subtest 'There is a linear delay for each hook script which exits with 142' => sub {
+            $job->settings->create({key => '_TRIGGER_JOB_DONE_RETRIES', value => '4'});
+            $hooks->{job_done_hook_failed} = 'echo delayed && exit 142;';
+            $job->update({state => UPLOADING});
+            $job->discard_changes;
+            $job->done;
+            perform_minion_jobs($t->app->minion);
+            $job_info = $t->app->minion->jobs({tasks => ['hook_script']})->next;
+            is_deeply(
+                $job_info->{args}[2],
+                {delay => 60, retries => 4, skip_rc => 142, kill_timeout => '30s', timeout => '5m'},
+                'args contain expected parameters'
+            );
+            is $job_info->{retries}, 1, 'hook script has been already tried once since has been queued';
+            is $job_info->{state}, 'inactive', 'job ends up inactive due to retry';
+            my $linear_delay = $job_info->{delayed} - $job_info->{retried};
+            is $linear_delay, 60, 'default delay for first retry applied correctly';
+            # force some retries with delay 0 to avoid having to wait in the test
+            $t->app->minion->job($job_info->{id})->retry({delay => 0}) for 1, 2;
+            perform_minion_jobs($t->app->minion);
+            $job_info = $t->app->minion->jobs({tasks => ['hook_script']})->next;
+            is $job_info->{retries}, 4, 'hook script has been retried the expected number of times';
+            $linear_delay = $job_info->{delayed} - $job_info->{retried};
+            is $linear_delay, 240, 'linear delay calculated correctly for 4th retry';
+            note "Retry one last time to validate the final state";
```

Comment:
I also wouldn't use `note` here, just a comment

Reply:
I will remove but I would like to just ask everyone why is this a problem when those are acting as comments? How can i achieve what I intend to?

Reply:
What does "those are acting as comments" mean?

Reply:
there are not visible, unless test run with verbose flag